-- unbundle the ruff binary so that it comes from devel/ruff
-- remove license lines to comply with tool versions that are currently in the tree

--- pyproject.toml.orig	2026-10-27 00:14:27 UTC
+++ pyproject.toml
@@ -1,6 +1,6 @@
 [build-system]
-requires = ["maturin>=1.9.3,<2.0"]
-build-backend = "maturin"
+requires = ["setuptools>0", "wheel"]
+build-backend = "setuptools.build_meta"
 
 [project]
 name = "ruff"
@@ -9,8 +9,6 @@ requires-python = ">=3.7"
 authors = [{ name = "Astral Software Inc.", email = "hey@astral.sh" }]
 readme = "README.md"
 requires-python = ">=3.7"
-license = "MIT"
-license-files = ["LICENSE"]
 keywords = [
     "automation",
     "flake8",
@@ -45,17 +43,6 @@ Changelog = "https://github.com/astral-sh/ruff/blob/ma
 Documentation = "https://docs.astral.sh/ruff/"
 Changelog = "https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md"
 
-[tool.maturin]
-bindings = "bin"
-manifest-path = "crates/ruff/Cargo.toml"
-module-name = "ruff"
-python-source = "python"
-strip = true
-exclude = [
-    "crates/ruff_linter/resources/test/fixtures/**/*",
-    "crates/ruff_linter/src/rules/*/snapshots/**/*"
-]
-
 [dependency-groups]
 dev = [
     "astral-dev-toolchain-cargo-codspeed>=5.0.1",
@@ -87,6 +74,10 @@ typeshed-formatting = [
 typeshed-formatting = [
     "black>=26.10.0",
 ]
+
+[tool.setuptools]
+packages = ["ruff"]
+package-dir = {"" = "python"}
 
 [tool.uv]
 no-build = true
